docs/source-fabric/advanced/model_parallel/tp.rst

```diff
@@ -0,0 +1,280 @@
+        # Use PyTorch's distributed tensor APIs to parallelize the model
+        plan = {
+            "w1": ColwiseParallel(),
+            "w2": RowwiseParallel(),
+            "w3": ColwiseParallel(),
+        }
+        parallelize_module(model, tp_mesh, plan)
+        return model
+
+Next, configure the :class:`~lightning.fabric.strategies.model_parallel.ModelParallelStrategy` in Fabric:
+
+.. code-block:: python
+
+    import lightning as L
+    from lightning.fabric.strategies import ModelParallelStrategy
+
+    # 1. Pass the parallelization function to the strategy
+    strategy = ModelParallelStrategy(parallelize_fn=parallelize_feedforward)
+
+    # 2. Configure devices and set the strategy in Fabric
+    fabric = L.Fabric(accelerator="cuda", devices=2, strategy=strategy)
+    fabric.launch()
+
+The strategy takes the custom parallelization function as input.
+No other changes to your training code are necessary at this point.
+Later in the code, when you call ``fabric.setup(model)``, Fabric will apply the ``parallelize_feedforward`` function to the model automatically.
+
+.. collapse:: Full training example (requires at least 2 GPUs).
+
+    .. code-block:: python
+
+        import torch
+        import torch.nn as nn
+        import torch.nn.functional as F
+
+        from torch.distributed.tensor.parallel import ColwiseParallel, RowwiseParallel
+        from torch.distributed.tensor.parallel import parallelize_module
+
+        import lightning as L
+        from lightning.pytorch.demos.boring_classes import RandomDataset
+        from lightning.fabric.strategies import ModelParallelStrategy
+
+
+        class FeedForward(nn.Module):
+            def __init__(self, dim, hidden_dim):
+                super().__init__()
+                self.w1 = nn.Linear(dim, hidden_dim, bias=False)
+                self.w2 = nn.Linear(hidden_dim, dim, bias=False)
+                self.w3 = nn.Linear(dim, hidden_dim, bias=False)
+
+            def forward(self, x):
+                return self.w2(F.silu(self.w1(x)) * self.w3(x))
+
+
+        def parallelize_feedforward(model, device_mesh):
+            # Lightning will set up a device mesh for you
+            tp_mesh = device_mesh["tensor_parallel"]
+            # Use PyTorch's distributed tensor APIs to parallelize the model
+            plan = {
+                "w1": ColwiseParallel(),
+                "w2": RowwiseParallel(),
+                "w3": ColwiseParallel(),
+            }
+            parallelize_module(model, tp_mesh, plan)
+            return model
+
+
+        strategy = ModelParallelStrategy(parallelize_fn=parallelize_feedforward)
+        fabric = L.Fabric(accelerator="cuda", devices=2, strategy=strategy)
+        fabric.launch()
+
+        # Initialize the model
+        model = FeedForward(8192, 8192)
+        model = fabric.setup(model)
+
+        # Define the optimizer
+        optimizer = torch.optim.AdamW(model.parameters(), lr=3e-3, foreach=True)
```

Comment:
Regarding 

> foreach=True

every time I read your tutorials, I learn something new 😊

Reply:
To be fair, this is copy-paste and not directly related to TP, so maybe it should actually be removed :))

Reply:
yes maybe this would lead readers to think that `foreach=True` is somehow related

Reply:
Yeah, I think it makes sense for this architecture in general, because it's not memory bound, but introducing it here could somehow make readers believe it is required or so. So maybe just dropping it would be better.

Reply:
removed 👍